test/katex-spec.js

```diff
@@ -1382,6 +1382,15 @@ describe("An HTML font tree-builder", function() {
         span = "<span class=\"mord mathbb\" style=\"color:blue;\">R</span>";
         expect(markup).toContain(span);
     });
+
+    it("should throw TypeError when the expression is of the wrong type", function() {
+        expect(function() {
+            katex.renderToString({badInputType: "yes"});
+        }).toThrowError(TypeError);
+        expect(function() {
+            katex.renderToString(["foo", {badInputType: "yes"}]);
+        }).toThrowError(TypeError);
+    });
```

Comment:
This only tests one of the two codepaths above, which will be OK if you move the checks to a single location.

What's the point of the second test? If it is the array, I'd avoid the nested object. But I don't see much need for a second object-based test. `null`, `undefined`, number or boolean would make for more interesting tests, but perhaps not interesting enough to actually do them.

Since your check you added assumes that `String` objects are valid input as well, we might add a test to verify that a `String` object as input does parse all right.